core/services/offchainreporting/run_saver.go

```diff
@@ -0,0 +1,70 @@
+package offchainreporting
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+type RunResultSaver struct {
+	utils.StartStopOnce
+
+	runResults     <-chan pipeline.RunWithResults
+	pipelineRunner pipeline.Runner
+	done           chan struct{}
+	jobID          int32
+}
+
+func NewResultRunSaver(runResults <-chan pipeline.RunWithResults, pipelineRunner pipeline.Runner, done chan struct{}, jobID int32) *RunResultSaver {
+	return &RunResultSaver{
+		runResults:     runResults,
+		pipelineRunner: pipelineRunner,
+		done:           done,
+		jobID:          jobID,
+	}
+}
+
+func (r *RunResultSaver) Start() error {
+	if !r.OkayToStart() {
+		return errors.New("cannot start already started run result saver")
+	}
+	go func() {
+		for {
+			select {
+			case rr := <-r.runResults:
+				logger.Debugw("saving job run", "run", rr.Run.ID, "task results", rr.TaskRunResults)
```

Comment:
Nit: can we label our logging e.g. `logger.Debugw("RunSaver: saving job run", "id", "run_saver" ...)`